stablehlo/reference/ProcessGrid.h

```diff
@@ -212,21 +255,44 @@ class ProcessGrid {
                                                      ProcessId processId,
                                                      const Tensor &operand);
 
+  /// Sends `inputs` to a channel with `channelId`.
+  /// The channel with `channel_id` is emptied before the receiving process can
+  /// receive values. If there are multiple processes sending data to a
+  /// duplciate `channelId`, the behavior is undefined.
+  void send(ArrayRef<Tensor> inputs, ChannelId channelId, ProcessId processId);
+
  private:
   /// StableHLO `num_replicas`.
   const uint32_t numReplicas_;
 
   /// StableHLO `num_partitions`.
   const uint32_t numPartitions_;
 
-  /// Interal queue of strings which represents `func::FuncOp` mnemonic that
+  /// Internal queue of strings which represents `func::FuncOp` mnemonic that
   /// returns a vector of Tensor. The function name is stored instead of the
   /// vector of tensors to save memory. See `ThreadSafeQueue`.
   detail::ThreadSafeQueue<StringAttr> infeed_;
```

Comment:
I thought we flipped this to `FlatSymbolRefAttrs`? Any reason to use string here?

Reply:
Mostly because `StringType` is a supported StableHLO type and `StringAttr` seems to be the best representation for this whereas `FlatSymbolRefAttrs` is more closely tied to MLIR related things. Since we now use `FlatSymbolRefsAttr` in the td file, function names are also verified.

Reply:
Valid reasoning! The benefit of a symbol ref vs string isn't too big while interpreting, so using StringType for the sake of it being spec'ed makes sense.